week-8/2-async.js

```diff
@@ -0,0 +1,18 @@
+// Turn getFlights() into an async function and use the .then(..) method to implement the following functions in order
```

Comment:
Sorry, Ali, not sure I understand the second part of your comment. I thought Promises were covered previously in JS2. I will be going over this exercise in class after covering `async`/`await` with the students. The intention here is to refactor the code so that they add an `async` keyword before the getFlightDetails() function to turn it into a Promise.

I admit, a lot of the stuff I've posted are looking to be very vague. I'm going to do another once over after the class to make sure the content is more clear.

Reply:
Ah ok, perhaps I'm misreading the comment then. I was thinking that we'd have to do something like:

```js
return new Promise((resolve, reject) => {
  if (promiseResolved) {
    resolve()
  } else {
    reject()
  }
})
```

This creates our own promise that we can decide when to resolve/reject. But it's a) not something I'd do very often and b) a bit confusing. Therefore in JS2 we avoid it (mostly).

However if we can solve just with the `async` keyword then that's fine 👍